web-registry/src/lib/denom.utils.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Convert Cosmos-style micro denom (starting with u) to natural denom.
+ * Example uregen -> REGEN
+ */
+export const formatDenomText = (askDenom: string): string => {
+  return askDenom.substring(1).toUpperCase();
+};
+
+/**
+ * Convert Cosmos-style micro denom amount to readable full-denom amount
+ * Example 7000000 uregen -> 7 REGEN
+ */
+export const microToDenom = (amount?: string | number): number => {
+  if (!amount) return 0;
+  const price = typeof amount === 'string' ? Number(amount) : amount;
+  return price / Math.pow(10, 6);
+};
+
+/**
+ * Convert readable full-denom amount to micro denom amount
+ * Example 8 REGEN -> 8000000 uregen
+ */
+export const denomToMicro = (amount?: string | number): number => {
+  if (!amount) return 0;
```

Comment:
I don’t know full context here, but should `amount` be a required field, or throw an error if it’s not present instead of returning 0?

Reply:
The idea here is that we can do less code-checking before invoking by handling the null case. 
Throwing an error is not great in this case, because there may be cases where it is temporarily undefined, but can be updated when data comes in.